ui/src/main/webapp/src/app/reports/jpa/jpa-report.service.ts

```diff
@@ -0,0 +1,45 @@
+import {Injectable} from "@angular/core";
+import {Http} from "@angular/http";
+import {Observable} from "rxjs/Observable";
+
+import {ProjectTechnologiesStatsModel} from "../../generated/tsModels/ProjectTechnologiesStatsModel";
+import {GraphService} from "../../services/graph.service";
+import {GraphJSONToModelService} from "../../services/graph/graph-json-to-model.service";
+import {Cached} from "../../shared/cache.service";
+import {JPAEntityModel} from "../../generated/tsModels/JPAEntityModel";
+import {JPANamedQueryModel} from "../../generated/tsModels/JPANamedQueryModel";
+import {JPAPersistenceUnitModel} from "../../generated/tsModels/JPAPersistenceUnitModel";
+import {JPAConfigurationFileModel} from "../../generated/tsModels/JPAConfigurationFileModel";
+
+@Injectable()
+export class JpaReportService extends GraphService
+{
+    constructor(http: Http, graphJsonToModelService: GraphJSONToModelService<any>) {
+        super(http, graphJsonToModelService);
+    }
+
+    @Cached('jpaReport', null, true)
+    getJpaInfo(execID: number): Observable<JpaReportInfo>
+    {
+        let a = this.getTypeAsArray<JPAEntityModel>(JPAEntityModel.discriminator, execID, {depth: 2, includeInVertices: false});
+        let b = this.getTypeAsArray<JPANamedQueryModel>(JPANamedQueryModel.discriminator, execID, {depth: 2, includeInVertices: false});
+        let c = this.getTypeAsArray<JPAPersistenceUnitModel>(JPAPersistenceUnitModel.discriminator, execID, {depth: 2, includeInVertices: false});
+        let d = this.getTypeAsArray<JPAConfigurationFileModel>(JPAConfigurationFileModel.discriminator, execID, {depth: 2, includeInVertices: false});
+        return Observable.forkJoin(a, b, c, d).map(val => {
```

Comment:
Why combining all fetches into 1 observable? Maybe you could use same approach as we use everywhere else and load them separately?

Now user would have to wait for all requests to finish, to see anything. But if you keep them separated, some requests would load first and user could already see some data, before all other requests finish.